test/core/gprpp/no_destruct_test.cc

```diff
@@ -0,0 +1,36 @@
+// Copyright 2022 gRPC authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include "src/core/lib/gprpp/no_destruct.h"
+
+#include <memory>
+
+#include <gtest/gtest.h>
+
+namespace grpc_core {
+namespace testing {
+namespace {
+
+NoDestruct<std::unique_ptr<int>> g_test_int(new int(42));
```

Comment:
drive-by suggestion: add a test to exercise some type with `MyType::~Mytype() { FAIL() << "Should never be destructed"; }`, at the very least for regression testing.

Reply:
https://github.com/grpc/grpc/pull/30497